File: mc_raylib/src/triangle.rs
use raylib::prelude::*;

#[derive(Clone, Copy)]
pub struct Triangle {
    pub vertices: [Vector3; 3],
    pub color: Color,
}
impl Triangle {
    pub fn new(v1: Vector3, v2: Vector3, v3: Vector3, color: Color) -> Self {
        Self {
            vertices: [v1, v2, v3],
            color,
        }
    }
}

File: mc_raylib_cpp/triangle.cpp
#include "raylib.h"

#include "include/triangle.h"


Triangle::Triangle(Vector3 v1, Vector3 v2, Vector3 v3, Color color) {
	vertices[0] = v1;
	vertices[1] = v2;
	vertices[2] = v3;
	this->color = color;
}